core/src/test/java/com/google/zxing/pdf417/decoder/PDF417DecoderTestCase.java

```diff
@@ -175,4 +181,197 @@ public void testSampleWithNoDataNoMacro() throws FormatException {
     DecodedBitStreamParser.decode(sampleCodes, "0");
   }
 
+  @Test
+  public void testUppercase() throws WriterException, FormatException {
+    //encodeDecode("", 0);
+    encodeDecode("A", 3);
+    encodeDecode("AB", 4);
+    encodeDecode("ABC", 5);
+    encodeDecode("ABCD", 6);
+    encodeDecode("ABCDE", 4);
+    encodeDecode("ABCDEF", 4);
+    encodeDecode("ABCDEFG", 5);
+    encodeDecode("ABCDEFGH", 5);
+  }
+
+  @Test
+  public void testNumeric() throws WriterException, FormatException {
+    encodeDecode("1", 2);
```

Comment:
I suppose you could write these type of tests with a loop, using substring of the entire string from 0 to N